corehq/apps/app_manager/models.py

```diff
@@ -4444,23 +4475,32 @@ def generate_shortened_url(self, url_type):
         else:
             return shortened_url
 
-    def get_short_url(self):
-        if not self.short_url:
-            self.short_url = self.generate_shortened_url('download_jad')
-            self.save()
-        return self.short_url
-
-    def get_short_odk_url(self, with_media=False):
-        if with_media:
-            if not self.short_odk_media_url:
-                self.short_odk_media_url = self.generate_shortened_url('download_odk_media_profile')
+    def get_short_url(self, build_profile_id=None):
+        if not build_profile_id:
+            if not self.short_url:
+                self.short_url = self.generate_shortened_url('download_jad')
                 self.save()
-            return self.short_odk_media_url
+            return self.short_url
         else:
-            if not self.short_odk_url:
-                self.short_odk_url = self.generate_shortened_url('download_odk_profile')
-                self.save()
-            return self.short_odk_url
+            return self.generate_shortened_url('download_jad', build_profile_id)
```

Comment:
fair enough. I suppose there's not much risk in generating a new URL each time since there is likely not much risk of hitting the limit of URLs we can generate (currently bit.ly I think).

fwiw, I wouldn't suggest creating new properties on the Application class for every possible URL. If the build profile has a short URL then that should be a property of the build profile.